### ISSUE TITLE
Retire addresses after confirmation, not before

### ISSUE THREAD

**Issue by imaginaryusername** [Electron-Cash/Electron-Cash#2638]
Related: https://github.com/Electron-Cash/Electron-Cash/issues/2026

An incident has happened where a change output address (from fusion) was retired when the parent tx was unconfirmed, leaving its child tx hanging at "Unconfirmed parent" despite both the parent and child tx having confirmed since. Switching servers and restarting the wallet did not help. Getting the address out of retirement, though, did help after a wallet restart: 

```
from electroncash.address import Address

wallet.synchronizer.change_scripthashes_that_are_retired.discard(Address.from_string('<paste_address_here>').to_scripthash_hex())
```

The suspicion here is retirement of this address happened in a problematic order - before its confirmation status can be updated - and so the child was left hanging without the possibility of auto-updating. It may be prudent for the retirement logic to additionally check if all transactions involving the address has been confirmed.

**Comment by cculianu** [comment]
Yes thank you very much for helping to investigate and report this issue. This needs fixing.
